buildpacks/ruby/src/layers/env_secret_key_base_layer.rs

```diff
@@ -0,0 +1,67 @@
+use crate::RubyBuildpack;
+use crate::RubyBuildpackError;
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+pub(crate) struct EnvSecretKeyBaseLayer;
+
+/// # Set the `SECRET_KEY_BASE` environment variable
+///
+/// This environment variable is primarially used for encrypting/decrypting user sessions.
+/// Developers should not need to set it themselves, however if they want,
+/// they can over-write the default.
+///
+/// Must run before any `rake` or `rails` commands are executed as the application may
+/// error without this environment variable.
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub(crate) struct EnvSecretKeyBaseLayerMetadata {
+    default_value: String,
+}
+
+impl Layer for EnvSecretKeyBaseLayer {
+    type Buildpack = RubyBuildpack;
+    type Metadata = EnvSecretKeyBaseLayerMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: true,
+            cache: true,
```

Comment:
I seem to remember (but can't find in the spec right now), that the cache is not guaranteed to be restored in all cases (eg the platform can choose not to, or to expire it). So I think this implementation is at risk of being fragile. It also means we have the same worry about never being able to mass purge caches, as we do for Heroku classic - when it would be great to get away from that situation.

It feels `store.toml` might be the CNB feature intended for long term storage like this?

Reply:
I'm not familiar with `SECRET_KEY_BASE`, but I assume this is used to encrypt sessions in cookies to they can't be tampered with on the client side? If so, if the cache would be dropped (like @edmorley mentioned) all user sessions will be invalidated. 

`store.toml` is meant as a persistent metadata store and probably, as @edmorley suggested, the better choice here. I have unresolved this conversation as I think we should at least have some sort of decision record what to do here - even if that decision is "we will still do it as a layer".

Reply:
I agree it should not be in a cache. This data needs to persist beyond many deploys and would force log everyone out if lost. I found store.toml related code here https://github.com/heroku/libcnb.rs/blob/b494b0baeef39015be7b5773402e903dfc9b3919/libcnb/src/runtime.rs#L197 so I see I can set it via a builder, but I'm not sure where I can read that data back in via a libcnb interface.

Reply:
Updated to use `store.toml` instead of the cache. Take a look. https://github.com/heroku/libcnb.rs/pull/531 would clean up the implementation a bit.